src/components/formBuilder/formBuilder.jsx

```diff
@@ -0,0 +1,280 @@
+import React, { useState } from 'react';
+import { connect } from 'react-redux';
+
+import { saveFormSchemaThunk } from '../../redux/forms-Reducer';
+import style from './formBuilder.module.css';
+
+function FormBuilder(props) {
+  const [state, setState] = useState({
+    inputs: [],
+    formName: '',
+    formDesc: ''
+  });
+  const [active, setActive] = useState('formName');
+  const [errors, setErrors] = useState({});
+
+  const onHandleChnage = (e, key) => {
+    const name = e.target.name;
+    const value =
+      e.target.type === 'checkbox' ? e.target.checked : e.target.value;
+
+    if (name !== 'formName' && name !== 'formDesc') {
+      const inputs = state.inputs;
+      inputs.map((item, index) => {
+        if (index === key) {
+          if (name === 'type' && item.type) {
+            Object.getOwnPropertyNames(item).forEach(function(val) {
+              if (val !== 'type') delete item[val];
+            });
+          }
+          item[name] = value;
+        }
+        return item;
+      });
+
+      setState({ ...state, inputs });
+    } else {
+      setState({ ...state, [name]: value });
+    }
+
+    control();
+  };
+
+  const control = () => {
+    let errors = { inputs: [] };
+    let control = true;
+
+    state.inputs.forEach((item, key) => {
+      if (!item.label) {
+        control = false;
+      } else errors.inputs[key] = {};
+    });
+
+    if (!state.formName) {
+      control = false;
+      errors.formName = 'Form name is required';
+    }
+    if (!state.formDesc) {
+      control = false;
+      errors.formDesc = 'Form description is required';
+    }
+    if (state.inputs.length === 0) {
+      errors.inputList = 'Inputs is required';
+      control = false;
+    }
+    if (control) return true;
+    else setErrors({ ...errors });
+    return false;
```

Comment:
If I'm not wrong you basically do this:
```suggestion
    !control && setErrors({ ...errors });
```